aptos-move/aptos-vm/src/zkid_validation.rs

```diff
@@ -38,11 +40,41 @@ fn get_jwks_onchain(resolver: &impl AptosMoveResolver) -> anyhow::Result<Patched
     Ok(jwks)
 }
 
+fn get_jwk_for_zk_authenticator(
+    jwks: &PatchedJWKs,
+    zkid_pub_key: &ZkIdPublicKey,
+    zkid_sig: &ZkIdSignature,
+) -> anyhow::Result<JWK, VMStatus> {
+    let jwt_header_parsed = zkid_sig.parse_jwt_header().map_err(|_| {
+        VMStatus::error(
+            StatusCode::INVALID_SIGNATURE,
+            Some("Failed to get JWT header".to_owned()),
+        )
+    })?;
+    let jwk_move_struct = jwks
+        .get_jwk(&zkid_pub_key.iss, &jwt_header_parsed.kid)
+        .map_err(|_| {
+            VMStatus::error(
+                StatusCode::INVALID_SIGNATURE,
+                Some("JWK not found".to_owned()),
+            )
+        })?;
+
+    let jwk = JWK::try_from(jwk_move_struct).map_err(|_| {
+        VMStatus::error(
+            StatusCode::INVALID_SIGNATURE,
+            Some("Could not parse JWK".to_owned()),
+        )
+    })?;
+    Ok(jwk)
+}
+
 pub fn validate_zkid_authenticators(
     transaction: &SignedTransaction,
     resolver: &impl AptosMoveResolver,
     _session: &mut SessionExt,
     _log_context: &AdapterLogSchema,
+    chain_id: ChainId,
 ) -> anyhow::Result<(), VMStatus> {
     // TODO(ZkIdGroth16Zkp): The ZKP/OpenID sig verification does not charge gas. So, we could have DoS attacks.
```

Comment:
cc @vgao1996 

We probably should not write such comments in public repo?😂 At least say that "we should charge" :)

Reply:
Actually, you cannot even charge because this is a prologue?

Reply:
This has not be enabled, right? If so then we are fine now.

Good to be cautious though!

Reply:
But it will be?😅

Reply:
Yeah just make it clear that proper gas metering needs to be implemented before we could expose this to our users.

Reply:
Will take care of this.